src/main/java/com/rakovets/course/java/core/practice/concurrency/Producer.java

```diff
@@ -0,0 +1,34 @@
+package com.rakovets.course.java.core.practice.concurrency;
+
+import java.util.Queue;
+import java.util.Scanner;
+
+public class Producer implements Runnable {
+    private Queue<Integer> integerNumbersQueue;
+
+    public Producer(QueueOfIntegerNumbers queueOfIntegerNumbers) {
+        integerNumbersQueue = queueOfIntegerNumbers.getQueue();
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            Scanner scanner = new Scanner(System.in);
+            System.out.print("Enter integer number: ");
+            try {
+                if (scanner.hasNextInt()) {
```

Comment:
What happens if the value is not a number?

Reply:
throw new UserInputException("Enter only integer number!");